global-indexes.md

```diff
@@ -0,0 +1,320 @@
+
+- **Before v7.6.0**: TiDB only supports local indexes on partitioned tables. This means that unique keys on partitioned tables have to include all columns in the partition expression. Queries that do not use the partition key have to scan all partitions, resulting in degraded query performance.
+- **[v7.6.0](https://docs.pingcap.com/tidb/stable/release-7.6.0)**: Introduces the [`tidb_enable_global_index`](/system-variables.md#tidb_enable_global_index-new-in-v760) system variable to enable global indexes. However, at that time the feature is still under development and is not recommended for production use.
+- **[v8.3.0](https://docs.pingcap.com/tidb/stable/release-8.3.0)**: Global indexes are released as an experimental feature. You can explicitly create a global index using the `GLOBAL` keyword when defining an index.
+- **[v8.4.0](https://docs.pingcap.com/tidb/stable/release-8.4.0)**: The global indexes feature becomes generally available (GA). You can create global indexes directly using the `GLOBAL` keyword without setting the `tidb_enable_global_index` system variable. From this version onward, the system variable is deprecated and fixed to `ON`, meaning global indexes are enabled by default.
+- **[v8.5.0](https://docs.pingcap.com/tidb/stable/release-8.5.0)**: Global indexes support including all columns from the partition expression.<!-- - **v9.0.0**: Global indexes support non-unique indexes. In partitioned tables, any index except clustered indexes can be created as a global index. -->
+
+## Global indexes vs. local indexes
+
+The following diagram shows the differences between global indexes and local indexes.
+
+<img src="https://github.com/hfxsd/docs/blob/global-index-best-practices/media/global-index-vs-local-index.png" alt="Global Index vs. Local Index" width="60%" height="60%"/>
+
+**Scenarios for global indexes**:
+
+- **Infrequent data archiving**: For example, in the healthcare industry, some business data must be retained for up to 30 years. Data is often partitioned monthly, creating 360 partitions at once, with very few `DROP` or `TRUNCATE` operations. In such scenarios, global indexes are more suitable, providing cross-partition consistency and improved query performance.
+- **Queries that require cross-partition data**: When queries need to access data across multiple partitions, global indexes can avoid full scans across all partitions and enhance query efficiency.
+
+**Scenarios for local indexes**:
+
+- **Frequent data archiving**: If data archiving operations are frequent and queries are mostly confined to a single partition, local indexes can offer better performance.
+- **Partition exchange requirements**: In industries like banking, processed data might first be written to a regular table and, after verification, exchanged into a partitioned table to minimize performance impact. In this case, local indexes are preferred, because enabling global indexes disables the partition exchange functionality on the table.
```

Comment:
```suggestion
- **Frequent data archiving**: If data archiving operations occur frequently and most queries are limited to a single partition, local indexes can offer better performance.
- **Use of partition exchange**: In industries like banking, processed data might be written to a regular table first and then exchanged into a partitioned table after verification, to minimize performance impact on the partitioned table. In this case, local indexes are preferred because once a global index is used, the partitioned table no longer supports partition exchange.
```